custom_code/hover_menu.js

```diff
@@ -0,0 +1,195 @@
+menu items. Namely, we can change menu items' styles based on user's mouse movements.
+This works because element style changes are caught by cobrowser and sent to the operator.
+
+Beware that this script is dependent on the structure of the menu and has been written
+solely based on the structure of a certain menu.
+**/
+
+// Helper function to query from direct children.
+var queryDirectChildren = function (parent, selector) {
+  if (!parent.children) return [];
+
+  var result = [];
+
+  for (var i = 0; i < parent.children.length; i++) {
+    var child = parent.children[i];
+    // `msMatchesSelector` is required for IE
+    var matches = child.matches || child.msMatchesSelector || undefined;
+    if (matches && matches.call(child, selector)) {
+      result.push(child);
+    }
+  }
+
+  return result;
+}
+
+// Function that styles for a sub-menu for it to be visible
+var expandMenu = function(subMenuContainerStyle, level, phoneMediaQueryList, right) {
+  subMenuContainerStyle.transform = 'scale(1, 1)';
+  subMenuContainerStyle.opacity = '1.0';
+
+  // Next steps are needed for setting sub-menu location,
+  // which depends on screen size, menu level and whether top menu is on the right
+
+
+  // First check if we're in phone view
+  if (phoneMediaQueryList.matches) {
+    if (level === 2) {
+      subMenuContainerStyle.left = '41px';
+    } else {
+      subMenuContainerStyle.left = '0';
+    }
+    // In phone view, level 1 menu items fill the whole space under menu bar
+    // and level 2 items are slightly to the right. No other styles need to be changed,
+    // so do an early return.
+    return;
+  }
+
+  // The location for level 1 sub-menu
+  if (level === 1) {
+    // This basically brings sub-menu to the screen.
+    // If not hovered, it's `-9999px;` to the left.
+    subMenuContainerStyle.left = '0';
+  // The location for level 2 sub-menu
+  } else {
+    if (right) {
+      // if top menu item is on the right, then level 2
+      // menu items are shown on the left side
+      subMenuContainerStyle.right = '100%';
+      subMenuContainerStyle.left = 'auto';
+    } else {
+      subMenuContainerStyle.left = '100%';
+    }
+  }
+}
+
+
+// Add mouse listeners to a menu item and its sub-items to change their style when hovered,
+// and restore old styles when not hovered anymore.
+var processMenuItem = function(menuItem, level, phoneMediaQueryList, right) {
+  // Sub-menu containers are direct children of a menu item
+  // and have either `grid-container3` or `grid-container7` class
+  var subMenuContainers = queryDirectChildren(menuItem, '.grid-container3, .grid-container7');
+
+  // Do nothing if menu item does not have dropdown
+  if (subMenuContainers.length === 0) return {restore: function() {}};
+
+  var subMenuContainer = subMenuContainers[0];
+  var subMenuContainerStyle = subMenuContainer.style;
+
+  // Expand menu when user hovers it
+  var mouseOverListener = menuItem.addEventListener('mouseover', function(event) {
```

Comment:
FYI `mouseover` bubbles to parent elements when child elements are entered, don't know if that changes anything.

Reply:
Yes, and it's expected.

When a user hovers a level 2 sub-menu, then she also hovers level 1 sub-menu. There are two listeners that handle the event - level 1 handler makes sure the first dropdown is expanded and level 2 handler makes sure that the second dropdown is expanded.